engines/private/symbol.cpp

```diff
@@ -0,0 +1,163 @@
+    s->u.val = v;
+}
+
+/* find s in symbol table symlist */
+Symbol *lookup(Common::String s, SymbolMap symlist)	 {
+    Symbol *r = symlist.getVal(s);
+    return r;
+}
+
+/* lookup some name in some symbol table */
+Symbol *lookupName(char *n) {
+    //debug("looking up %s", n);
+    Common::String *s = new Common::String(n);
+
+    if (settings.contains(*s))
+        return lookup(*s, settings);
+
+    else if (variables.contains(*s))
+        return lookup(*s, variables);
+
+    else if (cursors.contains(*s))
+        return lookup(*s, cursors);
+
+    else if (locations.contains(*s))
+        return lookup(*s, locations);
+
+    else if (rects.contains(*s))
+        return lookup(*s, rects);
+
+    else {
+        debug("WARNING: %s not defined", s->c_str());
+        return constant(STRING, 0, (char*) s->c_str());
+    }
+
+}
+
+
+void installAll(char *n) {
+    Common::String *s;
+    Common::Rect *r;
+
+    assert(stringToDefine.size() > 0);
+
+    while (!stringToDefine.empty()) {
+        s = new Common::String(stringToDefine.pop());
+        r = rectToDefine.pop();
+
+        //debug("name %s", s->c_str());
+        if (strcmp(n, "settings") == 0) {
+            assert(r == NULL);
+            install(s, STRING, 0, (char*) s->c_str(), r, &settings);
+        }
+
+        else if (strcmp(n, "variables") == 0) {
+            assert(r == NULL);
+            install(s, NAME, 0, NULL, r, &variables);
+            variableList.push_front(*s);
+        }
+
+        else if (strcmp(n, "cursors") == 0) {
+            assert(r == NULL);
+            install(s, NAME, 0, NULL, r, &cursors);
+        }
+
+        else if (strcmp(n, "locations") == 0) {
+            assert(r == NULL);
+            install(s, NAME, 0, NULL, r, &locations);
+            locationList.push_front(*s);
+        }
+
+        else if (strcmp(n, "rects") == 0) {
+            assert(r != NULL);
+            install(s, RECT, 0, NULL, r, &rects);
+        }
+        else
+            assert(0);
+
+    }
+
+}
+
+Symbol *constant(int t, int d, char *s) {
+    Symbol *sp;
+    Common::String *n = new Common::String("<constant>");
+
+    sp = (Symbol *) emalloc(sizeof(Symbol));
+    sp->name = n;
+    sp->type = t;
+    if (t == NUM || t == NAME)
+        sp->u.val = d;
+    else if (t == STRING)
+        sp->u.str = s;
+    else
+        assert(0);
+
+    constants.push_front(sp);
+    return sp;
+}
+
+/* install some symbol s in a symbol table */
+Symbol *install(Common::String *n, int t, int d, char *s, Common::Rect *r, SymbolMap *symlist) {
+    Common::String *name = new Common::String(*n);
+
+    Symbol *sp;
+
+    sp = (Symbol *) emalloc(sizeof(Symbol));
+    sp->name = name;
+    sp->type = t;
+    if (t == NUM || t == NAME)
+        sp->u.val = d;
+    else if (t == STRING)
+        sp->u.str = s;
+    else if (t == RECT)
+        sp->u.rect = r;
+    else
+        assert(0);
+
+    symlist->setVal(*n, sp);
+    assert(symlist->size() > 0);
+    return sp;
+}
+
+}
```

Comment:
I'd suggest to be consistent here and put everywhere `} // End of namespace Private`

Reply:
Fixed.